File: views/src/models/GenreModel.ts
export interface Genre {
    id: string;
    name: string;
    imageUrl: string | null;
    color: string | null;
    textColor: string | null;
    active: boolean;
    lastEdited: Date;
}

export interface genreCreateUpdate {
    name: string;
    image: File | null;
    color?: string;
    textColor?: string;
}

File: backend/src/main/java/com/castify/backend/models/genre/GenreModel.java
package com.castify.backend.models.genre;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class GenreModel {
    private String id;
    private String name;
    private String imageUrl;
    private boolean isActive;
    private LocalDateTime lastEdited;
}
